gradio/processing_utils.py

```diff
@@ -259,21 +268,33 @@ def save_file_to_cache(file_path: str | Path, cache_dir: str) -> str:
 
     return full_temp_file_path
 
+def check_public_url(url: str):
+    hostname = urlparse(url).hostname
```

Comment:
Btw sometimes Gradio is served on a url that is a raw ip address. Will this method work properly in that case? We should write unit tests for this function